src/auction/src/types.mo

```diff
@@ -0,0 +1,113 @@
+import AssocList "mo:base/AssocList";
+import List "mo:base/List";
+import RBTree "mo:base/RBTree";
+
+import Vec "mo:vector";
+
+import PriorityQueue "./priority_queue";
+
+module {
+
+  public type AssetId = Nat;
+  public type OrderId = Nat;
+
+  public type Account = {
+    // balance of user account
+    var credit : Nat;
+    // user's credit, placed as bid or ask
+    var lockedCredit : Nat;
+  };
+
+  public type Order = {
+    user : Principal;
+    userInfoRef : UserInfo;
+    assetId : AssetId;
+    price : Float;
+    var volume : Nat;
+  };
+
+  type AssetOrderBook_<O> = {
+    var queue : PriorityQueue.PriorityQueue<(OrderId, O)>;
+    var amount : Nat;
```

Comment:
The name `amount` can be confused with a token amount. Maybe length/size/nOrders/etc.

Reply:
replaced with "size"